pkg/agent/run.go

```diff
@@ -131,7 +127,7 @@ func coreClient(cfg string) (kubernetes.Interface, error) {
 	return kubernetes.NewForConfig(restConfig)
 }
 
-func Run(ctx context.Context, cfg cmds.Agent) error {
+func Run(ctx context.Context, cfg *cmds.Agent) error {
```

Comment:
Prior to this change we were passing by value, now we are passing by reference. Are there desired side-effects to the Agent struct var that Run and it's call tree perform for us that we cannot do explicitly prior to calling Run?

Reply:
this is because there are two go routines running in parallel:

goroutine #1 tries to get k3s/etcd/serveraddresses key from etcd to get the server ip and if it gets it it sets serverURLTmp 

goroutine #2 reads the serverURLtmp when its set and then set up the tunnel and start the agent with it

thats why I needed to pass by reference so that goroutine 2 gets an updated value when it set

Reply:
https://github.com/k3s-io/k3s/pull/2900#discussion_r572445480

Reply:
fixed